### DISCUSSION TITLE
SMS with Twilio Example - needs a fix?

### DISCUSSION THREAD

**Discussion by zburns** [postfixadmin/postfixadmin#618] [Q&A]
If you use Twilio SMS - I believe the code goes in the config.local.php.  It does not seem to work with postfixadmin when a user wants to reset the password - but the following code does work if I put it in a PHP stand-alone file and call it from the command line.  So I believe the code does work.  Just not sure why it would not work in postfixadmin.

Can anyone shed any light?
`
.
.
.
$CONF['sms_send_function'] = 'x_send_sms_twilio';

function x_send_sms_twilio($TO, $MESSAGE)
{
    $TWILIO_ACSID = "ABC123";
    $TWILIO_TOKEN = "DEF456";
    $URL = "https://api.twilio.com/2010-04-01/Accounts/" . $TWILIO_ACSID . "/Messages.json";
    $FROM = "1234567890";

    // Generated by curl-to-PHP: http://incarnate.github.io/curl-to-php/
    $ch = curl_init();

    curl_setopt($ch, CURLOPT_URL, $URL);
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
    curl_setopt($ch, CURLOPT_POST, 1);
    curl_setopt($ch, CURLOPT_POSTFIELDS, "Body=" . $MESSAGE . "&From=+1" . $FROM . "&To=+1" . $TO . "");
    curl_setopt($ch, CURLOPT_USERPWD, $TWILIO_ACSID . ':' . $TWILIO_TOKEN);

    $headers = array();
    $headers[] = 'Content-Type: application/x-www-form-urlencoded';
    curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);

    $result = curl_exec($ch);
    $sendback = true;

    if (curl_errno($ch))
    {
      //  echo 'Error:' . curl_error($ch);
      $sendback = false;
    }

    curl_close($ch);

    return $sendback;
}

.
.
.
`

**Comment by DavidGoodwin** [comment]
password-recover.php checks to see if there's something that's callable in the config property 'sms_send_function'; if there is it calls it ...
See 
https://github.com/postfixadmin/postfixadmin/blob/aa1b80d17391160705e9ac6cffd0db94d6ff3723/public/users/password-recover.php#L62

**Reply by DavidGoodwin** [reply]
but it will try and do an email first - https://github.com/postfixadmin/postfixadmin/blob/master/public/users/password-recover.php#L94

**Reply by zburns** [reply]
Thanks David - yep I get the email, but not the sms and have confirmed as you say it belongs in the config.local.php file.  Things look ok and my Twilio test works as a stand-alone snippet php program.  (Not a PHP programmer by day, but maybe somebody that uses Twilio can confirm/tweak my code to test also.)  Not sure if permissions issue or what.  Again - thanks!
